pkg/liquidity-mining/test/GaugeWorkingBalanceHelper.test.ts

```diff
@@ -0,0 +1,300 @@
+      args: [vault.address, votingEscrow.address, ZERO_ADDRESS],
+    });
+
+    boost = await deploy('VeBoostV2', { args: [proxyV1.address, votingEscrow.address] });
+
+    veDelegationProxy = await deploy('VotingEscrowDelegationProxy', {
+      args: [vault.address, votingEscrow.address, boost.address],
+    });
+  });
+
+  async function deployHelper(readTotalSupplyFromVE: boolean) {
+    workingBalanceHelper = await deploy('GaugeWorkingBalanceHelper', {
+      args: [veDelegationProxy.address, readTotalSupplyFromVE],
+    });
+  }
+
+  function itStoresParameters(readTotalSupplyFromVE: boolean) {
+    it('stores the veDelegationProxy', async () => {
+      expect(await workingBalanceHelper.getVotingEscrowDelegationProxy()).to.equal(veDelegationProxy.address);
+    });
+
+    it('stores the votingEscrow', async () => {
+      expect(await workingBalanceHelper.getVotingEscrow()).to.equal(votingEscrow.address);
+    });
+
+    it('indicates the network', async () => {
+      expect(await workingBalanceHelper.readsTotalSupplyFromVE()).to.equal(readTotalSupplyFromVE);
+    });
+  }
+
+  function itComputesWorkingBalances() {
+    const TOKENLESS_PRODUCTION = 0.4;
+    const MAX_BALANCE_RATIO = 1 / TOKENLESS_PRODUCTION;
+    const LOCK_PERIOD = 365 * DAY;
+    const stakeAmount = fp(5);
+    const bptAmount = fp(10);
+
+    async function createLockForUser(
+      account: SignerWithAddress,
+      amount: BigNumberish,
+      lockDuration: BigNumberish
+    ): Promise<void> {
+      await bpt8020.mint(account, amount);
+      await bpt8020.approve(votingEscrow, amount, { from: account });
+      const now = await currentTimestamp();
+      await votingEscrow.connect(account).create_lock(amount, now.add(lockDuration));
+    }
+
+    async function depositIntoGauge(account: SignerWithAddress, stakeAmount: BigNumberish) {
+      await pool.connect(admin).mint(account.address, stakeAmount);
+      await pool.connect(account).approve(gauge.address, stakeAmount);
+      await gauge.connect(account)['deposit(uint256)'](stakeAmount);
+    }
+
+    function currentBalanceEqualsProjected() {
+      it('projected balance should equal current', async () => {
+        const [currentWorkingBalance, projectedWorkingBalance] = await workingBalanceHelper.getWorkingBalances(
+          gauge.address,
+          user.address
+        );
+
+        // Ensure we have equal balances (that are non-zero)
+        expect(projectedWorkingBalance).to.eq(currentWorkingBalance);
+        expect(projectedWorkingBalance).to.gt(0);
+      });
+    }
+
+    function currentRatioEqualsProjected() {
+      it('current and projected ratios should be 1', async () => {
+        const [current, projected] = await workingBalanceHelper.getWorkingBalanceToSupplyRatios(
+          gauge.address,
+          user.address
+        );
+
+        // Ensure we have equal ratios (that equal 1)
+        expect(projected).to.eq(current);
+        expect(projected).to.eq(FP_ONE);
+      });
```

Comment:
This is the only ratio test we have - it'd be great to also test how the ratios change in a couple scenarios.